--- FRAMEWAVE_1.3.1_SRC/BuildTools/buildscripts/fwbuild.py.orig	2020-08-16 21:42:10.972877391 +0200
+++ FRAMEWAVE_1.3.1_SRC/BuildTools/buildscripts/fwbuild.py	2020-08-21 22:07:28.111946704 +0200
@@ -114,7 +114,7 @@
         self.oEnv['FWVARS'] = self.dctFwVars  # in case of changes to the internal dictionary
     	oProject = self.oEnv.SConscript( join( pthProjectDir, sProjectName+'.sconscript' ),\
                                          src_dir = pthProjectDir,\
-                                         build_dir = pthBuildDir,\
+                                         variant_dir = pthBuildDir,\
                                          duplicate = 0 )
     	if lstDependencies != None:
     		for dep in lstDependencies:
--- FRAMEWAVE_1.3.1_SRC/BuildTools/buildscripts/fwutil.py.orig	2023-03-14 18:13:43.493122918 +0100
+++ FRAMEWAVE_1.3.1_SRC/BuildTools/buildscripts/fwutil.py	2023-03-14 18:15:01.991235213 +0100
@@ -58,7 +58,7 @@ def setupIncludePaths ( oEnv, pthStartPa
     else:           lstAllHeaders = (os.path.dirname(f) for f in globtree(join( pthStartPath, "*.h" )))
 
     lstIncludes = []
-    if oEnv.has_key('CPPPATH'):
+    if 'CPPPATH' in oEnv:
         if ('#/'+pthStartPath) not in oEnv['CPPPATH']: lstIncludes = ['#/'+pthStartPath]
     else:
         oEnv.Append( CPPPATH= ['#/'+pthStartPath])
