--- FRAMEWAVE_1.3.1_SRC/Framewave/domain/fwImage/src/formatConversion.cpp~	2009-07-09 11:15:34.000000000 +0200
+++ FRAMEWAVE_1.3.1_SRC/Framewave/domain/fwImage/src/formatConversion.cpp	2021-08-17 20:02:38.764534982 +0200
@@ -11,22 +11,23 @@
 
 using namespace OPT_LEVEL;
 
-#define CMC_CHECK_PARAMS    if( pDst   ==0 || pSrc   ==0 )               return fwStsNullPtrErr;   \
-                            if( dstStep<=0 || srcStep<=0 )               return fwStsStepErr;      \
-                            if( roiSize.height<=0 || roiSize.width <=0 ) return fwStsSizeErr;
 
 FwStatus PREFIX_OPT( OPT_PREFIX, fwiYCbCr422_8u_C2P3R )( const Fw8u* pSrc, int srcStep, 
                                                            Fw8u* pDst[3],    int dstStep[3], 
                                                            FwiSize roiSize )
 {
-    CMC_CHECK_PARAMS; 
+    if( pDst   ==0 || pSrc   ==0 )               return fwStsNullPtrErr;
+    if( srcStep<=0 )               return fwStsStepErr;
+    if( roiSize.height<=0 || roiSize.width <=0 ) return fwStsSizeErr;
     return CMC::ModelCP <CMC::CH_FC_C2P3 >::ConvertREF( pSrc, srcStep, pDst, dstStep, roiSize ); 
 }
 
 FwStatus PREFIX_OPT( OPT_PREFIX, fwiYCbCr422_8u_P3C2R )( const Fw8u* const pSrc[3], int srcStep[3], 
                                                            Fw8u* pDst, int dstStep,   FwiSize roiSize )
 {
-    CMC_CHECK_PARAMS;
+    if( pDst   ==0 || pSrc   ==0 )               return fwStsNullPtrErr;
+    if( dstStep<=0 )               return fwStsStepErr;
+    if( roiSize.height<=0 || roiSize.width <=0 ) return fwStsSizeErr;
     return CMC::ModelPC< CMC::CH_FC_P3C2 >::ConvertREF( pSrc, srcStep, pDst, dstStep, roiSize ); 
 }
 
